cloudflare/src/framework/async_api.rs

```diff
@@ -42,12 +55,13 @@ impl Client {
             http_client,
         })
     }
+}
 
-    // Currently, futures/async fns aren't supported in traits, and won't be for a while.
-    // So unlike ApiClient/HttpApiClient, there's no trait with the request method.
-    pub async fn request<ResultType, QueryType, BodyType>(
+#[async_trait]
```

Comment:
I wonder if the language will ever support this directly - I wasn't able to find anything in the rust-lang issue tracker.

Reply:
The [async book](https://rust-lang.github.io/async-book/07_workarounds/06_async_in_traits.html) mentions it, and [this article explains why it's hard](https://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/)